Cache/Requests.lean

```diff
@@ -0,0 +1,102 @@
+import Cache.Hashing
+
+namespace Cache.Requests
+
+/-- Azure blob URL -/
+def URL : String :=
+  "https://lakecache.blob.core.windows.net/mathlib4"
+
+open System
+
+/-- Retrieves the azure token from the file system -/
+def getToken : IO String :=
+  return (← IO.FS.readFile ⟨"azure.token"⟩).trim
```

Comment:
```suggestion
  let some token ← IO.getEnv "AZURE_CACHE_SAS"
    | throw (IO.userError "environment variable AZURE_CACHE_SAS must be set to upload caches")
  return token
```
We usually pass these secrets as environment variables.

Reply:
Is this the simplest solution for mathematicians?

Reply:
I am confident that everybody who has upload privileges is able to set an environment variable.  In the beginning we're probably going to roll out upload only in CI like we do right now in mathlib3.